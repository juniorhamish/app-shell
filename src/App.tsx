import { Backdrop, Box, CircularProgress, Container } from '@mui/material';
import { useAuth0 } from '@auth0/auth0-react';
import { useTranslation } from 'react-i18next';
import AppBar from './components/AppBar';

function App() {
  const { t } = useTranslation();
  const { isLoading } = useAuth0();
  return (
    <>
      <Backdrop open={isLoading} aria-hidden={!isLoading} sx={(theme) => ({ zIndex: theme.zIndex.drawer + 1 })}>
        <CircularProgress color="inherit" aria-label={t('authentication-loading')} />
      </Backdrop>
      <Container maxWidth="lg" disableGutters aria-busy={isLoading}>
        <AppBar />
        <Box component="main" />
      </Container>
    </>
  );
}

export default App;
