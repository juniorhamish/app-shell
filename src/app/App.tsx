import { Backdrop, Box, CircularProgress, Container } from '@mui/material';
import { useTranslation } from 'react-i18next';
import AppBar from './features/app-bar/AppBar';
import Drawer from './features/drawer/Drawer';
import { UserInfoProvider } from '../context';
import useAuth0Listener from './features/auth/useAuth0Listener';
import { useAppSelector } from './hooks';
import { authState } from './features/auth/authSlice';

export default function App() {
  const { t } = useTranslation();
  useAuth0Listener();
  const { isLoading } = useAppSelector(authState);
  return (
    <UserInfoProvider>
      <Backdrop open={isLoading} aria-hidden={!isLoading} sx={(theme) => ({ zIndex: theme.zIndex.drawer + 1 })}>
        <CircularProgress color="inherit" aria-label={t('authentication-loading')} />
      </Backdrop>
      <Container maxWidth={false} disableGutters aria-busy={isLoading}>
        <AppBar />
        <Box component="main" />
        <Drawer />
      </Container>
    </UserInfoProvider>
  );
}
