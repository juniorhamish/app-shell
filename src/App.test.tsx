import { render, screen, within } from '@testing-library/react';
import App from './App';

describe('App', () => {
  describe('header bar', () => {
    it('should have a logo', () => {
      const { container } = render(<App />);

      expect(container).toMatchSnapshot();
    });
    it('should have a title', () => {
      render(<App />);

      expect(
        within(screen.getByRole('banner')).getByRole('heading'),
      ).toHaveTextContent('DAJohnston');
    });
    it('should have a Sign in button when the user is not currently logged in', () => {
      render(<App />);

      expect(
        within(screen.getByRole('banner')).getByRole('button', {
          name: 'Sign in',
        }),
      ).toBeVisible();
    });
  });
});
