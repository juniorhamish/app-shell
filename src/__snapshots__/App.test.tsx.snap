// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > header bar > should have a logo 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ptdwpq-MuiPaper-root-MuiAppBar-root"
      style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
      >
        <img
          alt=""
          class="MuiBox-root css-dks9xx"
          src="/src/assets/logo-round.png"
        />
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1bv5yfe-MuiTypography-root"
        >
          DAJohnston
        </h6>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Sign in
        </button>
      </div>
    </header>
    <main
      class="MuiBox-root css-0"
    />
  </div>
</div>
`;
